rotkehlchen/chain/evm/decoding/aura_finance/utils.py

```diff
@@ -0,0 +1,153 @@
+    update_cached_vaults(
+        display_name='Aura Finance',
+        database=evm_inquirer.database,
+        query_vaults=lambda: _query_aura_pools(evm_inquirer),
+        process_vault=_process_aura_pool,
+        cache_key=(CacheType.AURA_POOLS, str(evm_inquirer.chain_id.value)),
+    )
+
+
+def _process_aura_pool(database: 'DBHandler', vault: dict[str, Any]) -> None:
+    """Store Aura pool data.
+
+    May raise:
+    - NotERC20Conformant
+    """
+    chain_id = vault['chain_id']
+    aura_pool_token = get_or_create_evm_token(
+        userdb=database,
+        evm_address=vault['aura_pool_token'],
+        chain_id=chain_id,
+        encounter=TokenEncounterInfo(
+            description=f'Querying {chain_id.to_name()} aura finance balances',
+            should_notify=False,
+        ),
+        underlying_tokens=[
+            UnderlyingToken(
+                address=get_or_create_evm_token(
+                    userdb=database,
+                    chain_id=chain_id,
+                    evm_address=vault['underlying_bpt_token'],
+                    encounter=TokenEncounterInfo(
+                        description=f'Querying {chain_id.to_name()} aura finance balances',
+                        should_notify=False,
+                    ),
+                ).evm_address,
+                token_kind=EvmTokenKind.ERC20,
+                weight=ONE,
+            ),
+        ],
+        protocol=CPT_AURA_FINANCE,
+    )
+
+    if aura_pool_token.protocol != CPT_AURA_FINANCE:
+        log.debug(
+            f'Updating protocol for {chain_id.to_name()} aura finance '
+            f'asset {aura_pool_token}',
+        )
+        GlobalDBHandler.set_token_protocol_if_missing(
+            token=aura_pool_token,
+            new_protocol=CPT_AURA_FINANCE,
+        )
+
+
+def _query_aura_pools(evm_inquirer: 'EvmNodeInquirer') -> list[dict[str, Any]] | None:
+    """Query Aura pool data from the chain, returning None if the query fails."""
+    booster_contract = EvmContract(
+        address=CHAIN_ID_TO_BOOSTER_ADDRESSES[evm_inquirer.chain_id],
+        abi=AURA_BOOSTER_ABI,
+        deployed_block=0,  # unused in this context
+    )
+    try:
+        latest_pools_count = evm_inquirer.call_contract(
+            abi=booster_contract.abi,
+            method_name='poolLength',
+            contract_address=booster_contract.address,
+        )
+        results = evm_inquirer.multicall(
+            calls=[
+                (
+                    booster_contract.address,
+                    booster_contract.encode(
+                        method_name='poolInfo',
+                        arguments=[idx],
+                    ),
+                ) for idx in range(latest_pools_count)
+            ],
+        )
+    except RemoteError as e:
+        log.error(f'Failed to retrieve aura pools due to {e!s}')
+        return None
+
+    if len(results) == 0:
+        log.debug(f'No aura pools found on {evm_inquirer.chain_name}')
```

Comment:
add chain name here. It's in the inquirer

Reply:
it's already present in this particular diff or am i missing something?

Reply:
you just pushed it and force pushed in the PR

![2024-12-02_14-57](https://github.com/user-attachments/assets/e2629326-45f8-451c-a700-a753c1fb542e)